coreax/coresubset.py

```diff
@@ -425,27 +378,29 @@ def fit_to_size(self, coreset_size: int) -> None:
 
         :param coreset_size: The size of the of coreset to generate
         """
-        # Validate inputs
-        coreset_size = coreax.validation.cast_as_type(
-            x=coreset_size, object_name="coreset_size", type_caster=int
-        )
-        coreax.validation.validate_in_range(
-            x=coreset_size,
-            object_name="coreset_size",
-            strict_inequalities=True,
-            lower_bound=0,
-        )
-
         # Setup for sampling
         num_data_points = len(self.original_data.pre_coreset_array)
 
         # Randomly sample the desired number of points to form a coreset
-        random_indices = random.choice(
-            self.random_key,
-            a=jnp.arange(0, num_data_points),
-            shape=(coreset_size,),
-            replace=not self.unique,
-        )
+        try:
+            random_indices = random.choice(
+                self.random_key,
+                a=jnp.arange(0, num_data_points),
+                shape=(coreset_size,),
+                replace=not self.unique,
+            )
+        except AssertionError as exception:
```

Comment:
Where does the `AssertionError` come from? This would probably never be raised if python is run in the optimised mode that strips assert statements.

Reply:
I've updated my environment and this should actually be an AttributeError not an AssertionError - thanks for catching this - code updated